### ISSUE TITLE
api-workbench v0.8.40 fails to run in Atom v1.2.3 on Debian "Jessie"

### ISSUE THREAD

**Issue by tonedef71** [mulesoft/api-workbench#352]
After installing api-workbench v0.8.40, my Atom v1.2.3 editor is slow to start up.  It eventually reports the following error [ [Screenshot](https://snag.gy/vMjfIV.jpg) ]:

```
Error: Cannot find module 'electron'
Error: Cannot find module 'electron'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:286:25)
    at Module.require (module.js:365:17)
    at EventEmitter.<anonymous> (/usr/share/atom/resources/atom.asar/browser/lib/rpc-server.js:150:79)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)
    at EventEmitter.<anonymous> (/usr/share/atom/resources/atom.asar/browser/api/lib/web-contents.js:97:23)
    at emitTwo (events.js:87:13)
    at EventEmitter.emit (events.js:172:7)
    at metaToValue (/usr/share/atom/resources/atom.asar/renderer/api/lib/remote.js:105:15)
    at Object.exports.require (/usr/share/atom/resources/atom.asar/renderer/api/lib/remote.js:189:34)
    at Object.<anonymous> (/home/developer/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/fileDialogUtils.js:3:21)
    at Module._compile (module.js:434:26)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:190:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/developer/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:3337:19)
    at Module._compile (module.js:434:26)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:190:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/developer/.atom/packages/api-workbench/main.js:2:31516)
    at e (/home/developer/.atom/packages/api-workbench/main.js:1:200)
    at Object.module.exports.t (/home/developer/.atom/packages/api-workbench/main.js:1:5246)
    at e (/home/developer/.atom/packages/api-workbench/main.js:1:200)
    at Object.<anonymous> (/home/developer/.atom/packages/api-workbench/main.js:1:325)
    at e (/home/developer/.atom/packages/api-workbench/main.js:1:200)
    at /home/developer/.atom/packages/api-workbench/main.js:1:287
    at Object.<anonymous> (/home/developer/.atom/packages/api-workbench/main.js:1:292)
    at Module._compile (module.js:434:26)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:190:21)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Package.module.exports.Package.requireMainModule (/usr/share/atom/resources/app.asar/src/package.js:661:34)
    at /usr/share/atom/resources/app.asar/src/package.js:115:28
    at Package.module.exports.Package.measure (/usr/share/atom/resources/app.asar/src/package.js:92:15)
    at Package.module.exports.Package.load (/usr/share/atom/resources/app.asar/src/package.js:106:12)
    at PackageManager.module.exports.PackageManager.loadPackage (/usr/share/atom/resources/app.asar/src/package-manager.js:434:14)
    at PackageManager.module.exports.PackageManager.activatePackage (/usr/share/atom/resources/app.asar/src/package-manager.js:511:30)
    at /usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:444:29
    at exit (/usr/share/atom/resources/app.asar/node_modules/settings-view/lib/package-manager.js:73:16)
    at triggerExitCallback (/usr/share/atom/resources/app.asar/src/buffered-process.js:213:47)
    at /usr/share/atom/resources/app.asar/src/buffered-process.js:227:18
    at Socket.<anonymous> (/usr/share/atom/resources/app.asar/src/buffered-process.js:98:18)
    at emitOne (events.js:82:20)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

**Comment by tonedef71** [comment]
The npm version on my system is 1.4.3:
```
~$ npm --version
1.4.3
```
There is no module named "electron" in folder "~/.atom/packages/api-workbench/node_modules".

Running "apm install -d" from the "~/.atom/packages/api-workbench/node_modules" folder makes no difference:
```
~/.atom/packages/api-workbench/node_modules$ apm install -d
Installing modules ✓
~/.atom/packages/api-workbench/node_modules$ ls -al 
drwxr-xr-x 33 developer developer 4096 Nov 15 13:34 .
drwxr-xr-x  9 developer developer 4096 Nov 15 13:34 ..
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 arrify
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 atom-package-deps
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 atom-space-pen-views
drwxr-xr-x  4 developer developer 4096 Nov 15 13:34 atom-ui-lib
drwxr-xr-x  4 developer developer 4096 Nov 15 13:34 basarat-text-buffer
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 .bin
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 classnames
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 client-oauth2
drwxr-xr-x  4 developer developer 4096 Nov 15 13:34 emissary
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 http-response-object
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 invariant
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 jquery
drwxr-xr-x  8 developer developer 4096 Nov 15 13:34 jsonix
drwxr-xr-x  6 developer developer 4096 Nov 15 13:34 know-your-http-well
drwxr-xr-x  6 developer developer 4096 Nov 15 13:34 marked
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 mime
drwxr-xr-x  6 developer developer 4096 Nov 15 13:34 mkdirp
drwxr-xr-x  4 developer developer 4096 Nov 15 13:34 popsicle
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 pretty-data
drwxr-xr-x  9 developer developer 4096 Nov 15 13:34 raml-1-parser
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 raml-actions
drwxr-xr-x 13 developer developer 4096 Nov 15 13:34 raml-suggestions
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 react
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 react-pure-render
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 underscore
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 w3c-schemas
drwxr-xr-x  5 developer developer 4096 Nov 15 13:34 xhr2
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 xmldom
drwxr-xr-x  3 developer developer 4096 Nov 15 13:34 xmlhttprequest
drwxr-xr-x  2 developer developer 4096 Nov 15 13:34 xtend
drwxr-xr-x  4 developer developer 4096 Nov 15 13:34 yaml-ast-parser
```

**Comment by ddenisenko** [comment]
@dreamflyer thoughts?

**Comment by dreamflyer** [comment]
Looks strange. Electron is a core part of atom and it shouldn't be installed via apm or npm. Looks like electron isn't fully loaded yet when we trying to access it on workbench startup. So we will try to make electron accessing lazy, it should help.

**Comment by dreamflyer** [comment]
will be published in v0.8.41
